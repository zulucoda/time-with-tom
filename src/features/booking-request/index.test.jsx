import { render } from '@testing-library/react';
import BookingRequest from '.';
import React from 'react';
import {
  DateAndTimePickerProviderWrapperTestUtil,
  ReactRouterWrapperTestUtil,
} from '../../test-util';

describe('Features - Booking Request - Unit Test', function () {
  test('does it render', () => {
    const { container } = render(
      <ReactRouterWrapperTestUtil>
        <DateAndTimePickerProviderWrapperTestUtil>
          <BookingRequest />
        </DateAndTimePickerProviderWrapperTestUtil>
      </ReactRouterWrapperTestUtil>,
    );
    expect(container).toMatchSnapshot();
  });
});
