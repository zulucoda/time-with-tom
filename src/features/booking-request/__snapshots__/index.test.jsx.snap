// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features - Booking Request - Unit Test does it render 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      >
        <div
          class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom MuiTypography-alignCenter"
            marked="center"
          >
            Booking Request
          </h3>
          <form
            action="#"
          >
            <div
              class="sc-AxirZ fOBhug"
            >
              <div
                class="sc-AxjAm bLetVr"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
                >
                  activity
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-component-select-activity"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="mui-component-select-activity"
                    role="button"
                    tabindex="0"
                  >
                    Please select
                  </div>
                  <input
                    name="activity"
                    type="hidden"
                    value="none"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxhCb fVmqxl"
            >
              <div
                class="sc-AxiKw grOufF"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    name
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="sc-AxiKw grOufF"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    surname
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="surname"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="sc-AxiKw grOufF"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    email
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="email"
                      type="email"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-AxheI dLDGYA"
            >
              <div
                class="sc-AxgMl eTFDCq"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                  format="HH:mm"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    startTime
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="startTime"
                      placeholder="18:00"
                      readonly=""
                      type="text"
                      value="02:00"
                    />
                  </div>
                  
                </div>
              </div>
              <div
                class="sc-AxgMl eTFDCq"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                  format="HH:mm"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    endTime
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="endTime"
                      placeholder="18:00"
                      readonly=""
                      type="text"
                      value="02:00"
                    />
                  </div>
                  
                </div>
              </div>
              <div
                class="sc-AxhUy hqJxes"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                  format="yyyy/MM/dd"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    date
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="date"
                      placeholder="2020/04/21"
                      readonly=""
                      type="text"
                      value="2020/04/21"
                    />
                  </div>
                  
                </div>
              </div>
            </div>
            <pre>
              debug value
              {
   "activity": "none",
   "name": "",
   "surname": "",
   "email": "",
   "date": "2020-04-21T00:00:00.000Z",
   "startTime": "2020-04-21T00:00:00.000Z",
   "endTime": "2020-04-21T00:00:00.000Z"
}
            </pre>
            <pre>
              debug errors
              {}
            </pre>
            <button
              type="submit"
            >
               Submit 
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
